### PR TITLE
AV-840: file analysis status

### PR DESCRIPTION
**Pull request by bzar**
- Added new plugin ResourceStatusPlugin gor handling `resource_status` API action
- Added fields `sha256`, `malware_check` and `status_updated` to resource data model
- Added `admin_only_field` validator for package/resource fields

### CODE REVIEW

File: modules/ckanext-ytp_main/ckanext/ytp/validators.py [hunk 1]
```diff
@@ -459,3 +459,21 @@ def check_deprecation(key, data, errors, context):
...
+
+    def validator(key, data, errors, context):
+
+        if 'package' not in context:
+            return
+
+        data_dict = flatten_dict(get_action('package_show')(context, {'id': context['package'].id}))
+
+        if not authz.is_sysadmin(context['user']):
+            data[key] = data_dict[key]
```
Comment (Aaronkala) [inline modules/ckanext-ytp_main/ckanext/ytp/validators.py]: What does this do? It doesn't modify the data['key'] (so the field) if the user is a sysadmin? The update here is to keep the old value?

Comment (bzar) [reply modules/ckanext-ytp_main/ckanext/ytp/validators.py]: Yes. If the user is not an admin, we revert whatever change was made.

File: modules/ckanext-ytp_main/ckanext/ytp/schemas/dataset.json [hunk 2]
```diff
@@ -408,6 +408,21 @@
...
+      "validators": "ignore_missing admin_only_field"
+    },
+    {
+      "field_name": "sha256",
+      "form_snippet": "hidden.html",
+      "validators": "ignore_missing admin_only_field"
+    },
+    {
+      "field_name": "malware_check",
+      "form_snippet": "hidden.html",
```
Comment (Aaronkala) [inline modules/ckanext-ytp_main/ckanext/ytp/schemas/dataset.json]: Is the purpose to keep them hidden on the page or just not display them at all? I think if you pass null to display_snippet you can remove a field from being displayed and being posted at all. (https://github.com/ckan/ckanext-scheming#display_snippet).

Or is the purpose to keep it and just hide it?

File: modules/ckanext-ytp_main/ckanext/ytp/validators.py (hunk 1 above)
Comment (Aaronkala) [reply modules/ckanext-ytp_main/ckanext/ytp/validators.py]: Ah yes

File: modules/ckanext-ytp_main/ckanext/ytp/schemas/dataset.json (hunk 2 above)
Comment (bzar) [reply modules/ckanext-ytp_main/ckanext/ytp/schemas/dataset.json]: Yeah, that actually needs a bit more work. SHA256 and malware check result should show up on the page, but the `status_updated` not necessarily. I'll focus on the UI side more once the action part works.

Comment (bzar) [reply modules/ckanext-ytp_main/ckanext/ytp/schemas/dataset.json]: Show up on the view page, but not in the edit view, I mean.

**Pull request merged**